src/migrations.zig

```diff
@@ -0,0 +1,65 @@
+const std = @import("std");
+
+const main = @import("main.zig");
+const ZonElement = @import("zon.zig").ZonElement;
+const Palette = @import("assets.zig").Palette;
+
+var arenaAllocator = main.utils.NeverFailingArenaAllocator.init(main.globalAllocator);
+const migrationAllocator = arenaAllocator.allocator();
+
+var blockMigrations: std.StringHashMap([]const u8) = .init(migrationAllocator.allocator);
+
+pub fn registerBlockMigrations(migrations: *std.StringHashMap(ZonElement)) void {
+	std.log.info("Registering {} block migrations", .{migrations.count()});
+
+	var migrationIterator = migrations.iterator();
+	while(migrationIterator.next()) |migration| {
+		register(&blockMigrations, "block", migration.key_ptr.*, migration.value_ptr.*);
+	}
+}
+
+fn register(
+	collection: *std.StringHashMap([]const u8),
+	assetType: []const u8,
+	name: []const u8,
+	migrationZon: ZonElement,
+) void {
+	if((migrationZon != .object or migrationZon.object.count() == 0)) {
+		return;
+	}
+
+	var migrationZonIterator = migrationZon.object.iterator();
+	while(migrationZonIterator.next()) |migration| {
+		if(collection.contains(migration.key_ptr.*)) {
```

Comment:
There is also a `getOrPut` command which might avoid some redundant operations here.